### PR TITLE
Update NodeJS packaging patterns.

### PR DESCRIPTION
**Pull request by jmuk**
- retire use_pbjs flag -- it's not used anymore, and it does not
  fit with the new pattern of Gapic code.
- update the grpc package template: this fits with the proto packages
  used by gcloud-node.
- update gax package template: this will work well with the new
  pattern of Gapic code, as you can see in googleapis/toolkit#392
  or GoogleCloudPlatform/gcloud-node#1476

### CODE REVIEW

File: templates/nodejs/index.js [hunk 1]
```diff
@@ -14,6 +14,9 @@
...
 
-var grpc = require('grpc');
-exports.client = grpc.loadObject(require('./service'));
-exports.auth = require('google-auth-library');
+var path = require('path');
+
+module.exports = function(relative) {
+  return path.join(__dirname, 'proto', relative);
+};
+l
```
Comment (geigerj) [inline templates/nodejs/index.js]: typo on line 22?

**Comment by geigerj** [comment]
Good to see so much dead code being removed!

Please check failing test and one small note. Otherwise looks good.

File: templates/nodejs/index.js (hunk 1 above)
Comment (jmuk) [inline templates/nodejs/index.js]: right. removed.

**Comment by geigerj** [comment]
LGTM

**Pull request merged**